### PR TITLE
added test for shipCreator class

### PR DESCRIPTION
**Pull request by rivalez**

### CODE REVIEW

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/Board.java [hunk 1]
```diff
@@ -48,19 +48,34 @@ private void createCellInRow(EventHandler<? super MouseEvent> handler, HBox row,
...
+        if (canPlaceShip(ship, cells)) {
+            placeShip(ship, cells);
+            return true;
         }
         out.println(positions);
         return false;
     }
 
-    private boolean placeShip(Ship ship, Cell cell) {
+    boolean isShipPositionValid(Ship ship, Point2D cells) {
```
Comment (Ozymadias) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/Board.java]: why cells instead of cell?

File: battleship-client/src/test/java/com/epam/solid/nie/client/ui/ShipCreatorTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        ShipFactory shipFactory = new HorizontalShipFactory();
+        ShipCreator shipCreator = new ShipCreator(shipFactory);
+
+        List<Cell> cells = Arrays.asList(new Cell(Point2D.of(0, 0)), new Cell(Point2D.of(0, 1)), new Cell(Point2D.of(3, 0)), new Cell(Point2D.of(1, 1)));
+
+        //when
+        Ship ship = shipCreator.createShip(cells);
+
+        //then
+        Assert.assertTrue(ship.getBattleShip() instanceof HorizontalShip);
```
Comment (Ozymadias) [inline battleship-client/src/test/java/com/epam/solid/nie/client/ui/ShipCreatorTest.java]: what is the reason of this assertion?

File: battleship-client/src/test/java/com/epam/solid/nie/client/ui/ShipTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        ship.hit(Point2D.of(0,1));
+        ship.hit(Point2D.of(3,0));
+        ship.hit(Point2D.of(1,1));
+        ship.hit(Point2D.of(1,1));
+
+        //then
+        Assert.assertEquals(ship.getRemainingHealth(), 0);
+    }
+
+    public void should_TellIfShipIsAlive(){
```
Comment (Ozymadias) [inline battleship-client/src/test/java/com/epam/solid/nie/client/ui/ShipTest.java]: shouldn't there be another test of this kind to test whether method return true in appropriate case

File: battleship-client/src/test/java/com/epam/solid/nie/client/ui/ShipTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Test
+public class ShipTest {
+
+    public void should_removeHealth_whenHit(){
```
Comment (Ozymadias) [inline battleship-client/src/test/java/com/epam/solid/nie/client/ui/ShipTest.java]: maybe there should be also test if it got none-zero health after creation

File: common/src/main/java/com/epam/solid/nie/ships/AbstractShipPositions.java [hunk 5]
```diff
@@ -18,4 +19,17 @@
     void setPositions(List<Point2D> positions) {
         this.positions = positions;
     }
+
+    public BattleShipType getBattleShipType() {
+        return battleShipType;
+    }
+
+    public List<Point2D> getShipProperties() {
```
Comment (Ozymadias) [inline common/src/main/java/com/epam/solid/nie/ships/AbstractShipPositions.java]: why method called get if it is not a getter, also why shipProperties if its return position?
